seismostats/analysis/estimate_a.py

```diff
@@ -102,6 +103,13 @@ def estimate_a_classic(magnitudes: np.ndarray,
     """
     if delta_m is None:
         delta_m = 0
+        tolerance = 1e-08
+    else:
+        tolerance = max(delta_m / 100, 1e-08)
+    assert (
```

Comment:
does this have to be an assertion? wouldn't a warning be enough? I could see that I want to try stuff with the wrong binning on purpose to see the effect

Reply:
Hm, I think this is not as important as in the b-value estimation, however, the mc cutting will just be incorrect if the binning is incorrect. Then, isn't it best to not allow it? I don't think that there is much to be gained by trying to use the incorrect binnning..

Reply:
Final decision: the binning_testing function now only gives an error, if the binning could be larger than delta_m

Reply:
This makes my head hurt.

1. Currently this implementation does not give an error if the binning could be larger, but a warning. Also, it gives an error if it could be smaller. I guess that's what you meant, but you wrote something else.

2. You handle the warning inside binning test, but the error outside binning test. It only gives a warning if the fail_if_larger is False. I would suggest, you raise errors and warnings all inside the binning test and don't return anything OR you only return true or false, rename the flag "false_if_larger", (fail often implies that an error is raised), and handle the warnings and errors outside the binning_test.